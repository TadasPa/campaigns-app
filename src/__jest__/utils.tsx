import React, { PropsWithChildren, ReactElement } from "react";
import { render } from "@testing-library/react";
import { ThemeProvider, DefaultTheme } from "styled-components";
import { theme } from "../theme";
import GlobalStyles from "../theme/globalStyles";

const Providers = (initialTheme: DefaultTheme) => ({
  children,
}: PropsWithChildren<any>) => (
  <ThemeProvider theme={initialTheme}>
    <GlobalStyles />
    {children}
  </ThemeProvider>
);

interface ICustomRenderOptions {
  initialTheme?: DefaultTheme;
}

const customRender = (
  ui: ReactElement,
  { initialTheme = theme as any, ...restOptions }: ICustomRenderOptions = {}
) =>
  render(ui, {
    wrapper: Providers(initialTheme),
    ...restOptions,
  });

export * from "@testing-library/react";

export { customRender as render };
