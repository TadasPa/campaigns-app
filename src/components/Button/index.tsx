import React, { FC, HTMLAttributes } from "react";
import styled from "styled-components";
import { space, SpaceProps } from "styled-system";

type IButton = SpaceProps & HTMLAttributes<HTMLButtonElement>;

const Button: FC<IButton> = ({ onClick, children, ...rest }) => {
  return (
    <button onClick={onClick} {...rest}>
      {children}
    </button>
  );
};

const StyledButton = styled(Button)<IButton>`
  padding: 7px;
  width: 300px;
  border: 0;
  border-radius: 3px;
  background-color: ${({ theme: { colors } }) => colors.buttons.primary};
  color: ${({ theme: { colors } }) => colors.typography.light};
  font-family: ${({ theme: { textStyles } }) => textStyles.small.fontFamily};
  font-size: ${({ theme: { textStyles } }) => textStyles.small.fontSize};
  line-height: ${({ theme: { textStyles } }) => textStyles.small.lineHeight};

  &:hover {
    cursor: pointer;
  }

  ${space};
`;

export default StyledButton;
