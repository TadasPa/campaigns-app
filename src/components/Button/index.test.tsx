import React from "react";
import Button from ".";
import { render, screen } from "../../__jest__/utils";
import userEvent from "@testing-library/user-event";

test("Button machtes snapshot", () => {
  const { container } = render(<Button />);
  expect(container).toMatchSnapshot();
});

test("Button onClick callback is called", () => {
  const onClickCallback = jest.fn();
  render(<Button onClick={onClickCallback} />);

  userEvent.click(screen.getByTestId("button-component"));

  expect(onClickCallback).toHaveBeenCalledTimes(1);
});
