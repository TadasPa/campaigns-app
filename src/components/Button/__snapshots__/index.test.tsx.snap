// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button machtes snapshot 1`] = `
.c0 {
  padding: 7px;
  width: 300px;
  border: 0;
  border-radius: 3px;
  background-color: #3149b9;
  color: #FFFFFF;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.6;
}

.c0:hover {
  cursor: pointer;
}

<div>
  <button
    class="c0"
    data-testid="button-component"
  />
</div>
`;
